ForumSessionProvider/ForumSessionProvider.php

```diff
@@ -18,284 +18,851 @@
+
+	// Other Variables.
+	protected $loaded = false;
+	protected $logger;
+
+	/*
+	 * Load up all Forum Software Settings using the standard way.
+	*/
+	public function loadFSSettings()
+	{
+		global $wgSMFPath, $wgFSDebug, $wgFSPDenyGroups, $wgFSPAllowGroups, $wgFSPAdminGroups, $wgFSPSuperGroups, $wgFSPSpecialGroups;
+
+		$this->FSSettings['path'] = isset($wgSMFPath) ? $wgSMFPath : '../forum';
+		$this->FSSettings['debug'] = !empty($c) ? true : false;
+
+		// Some standard groups.
+		foreach (array(
+			'LoginDeniedGroups' => 'wgFSPDenyGroups',
+			'LoginAllowedGroups' => 'wgFSPAllowGroups',
+			'AdminGroups' => 'wgFSPAdminGroups',
+			'SuperAdminGroups' => 'wgFSPSuperGroups',
+			'SpecialGroups' => 'wgFSPSpecialGroups',
+			'NameStyle' => 'wgFSPNameStyle',
+		) as $key => $value)
+			$this->FSSettings[$key] = !empty($$value) ? $$value : array();
+	}
+
+	/*
+	 *	Combatiblity layer for Auth_SMF.php extension settings.
+	*/
+	public function combatAuthSMF()
+	{
+		global $wgSMFPath, $wgSMFDenyGroupID, $wgSMFGroupID, $wgSMFAdminGroupID, $wgSMFAdminGroupID, $wgSMFSpecialGroups;
+
+		$this->FSSettings['path'] = isset($wgSMFPath) ? $wgSMFPath : '../forum';
+		//$wgSMFLogin is not required.
+		//$wgCookieDomain is not required.
+
+		// Some standard groups.
+		foreach (array(
+			'LoginDeniedGroups' => 'wgSMFDenyGroupID',
+			'LoginAllowedGroups' => 'wgSMFGroupID',
+			'AdminGroups' => 'wgSMFAdminGroupID',
+			'SuperAdminGroups' => 'wgSMFAdminGroupID',
+			'SpecialGroups' => 'wgSMFSpecialGroups',
+			'NameStyle' => 'wgFSPNameStyle',
+		) as $key => $value)
+			$this->FSSettings[$key] = !empty($$value) ? $$value : $this->FSSettings[$key];
+
+		// Set the login style to SMF
+		if (empty($this->FSSettings['NameStyle']))
+			$this->FSSettings['NameStyle'] = 'smf';
+	}
+
+	/*
+	 * Read the Forum Settings file and load up settings for later on.
+	*/
+	public function readFSFSettings($file)
+	{
+		global $cookiename, $boardurl, $db_prefix, $db_type, $db_server, $db_user, $db_passwd, $db_name;
+
+		require ($this->FSSettings['path'] . '/Settings.php');
+
+		// Put these away.
+		foreach (array(
+			'cookiename',
+			'boardurl',
+			'db_prefix',
+			'db_type',
+			'db_server',
+			'db_user',
+			'db_passwd',
+			'db_name',
+			'auth_secret',
+			'cookie_no_auth_secret',
+			'image_proxy_secret',
+			'sourcedir'
+		) as $key)
+			$this->FSSettings[$key] = !empty($$key) ? $$key : null;
+	}
+
+	public function __construct(array $params = array())
+	{
+		global $wgSMFLogin, $wgFSPSoftware;
+
 		parent::__construct($params);
 
 		// TODO: Keep eye out for other things that need logging. Admin changes?
 		$this->logger = LoggerFactory::getInstance('ForumSessionProviderLog');
 
-        // Administrator always get admin rights.
-        if (!in_array(1, $GLOBALS['wgFSPAdminGroups']))
-            $GLOBALS['wgFSPAdminGroups'][] = 1;
+		// Set our software up.
+		$this->wgFSPSoftware = !empty($wgSMFLogin) ? 'smf2.0' : (!empty($wgFSPSoftware) ? $wgFSPSoftware : null);

```

Comment:
This is entirely for compatiblity with the old settings from Auth_SMF plugin.  If that setting exists, we enforced SMF 2.0 (as the plugin was technically not for 2.1) and continue on.

Reply:
No, I mean wgFSPSoftware isn't a defined class property in your revision. So I'm not sure why you're referencing it as such rather than just using the $wgFSPSoftware global.

Reply:
Ahh yea, I see what you mean.  Will add that to a protected.